cpp/src/core/core_number.cu

```diff
@@ -0,0 +1,144 @@
+/*
```

Comment:
Can we rename `core` directory into `cores`? Mostly because `src/core` can be misleading as one would expect "core" files for the whole project to live there. As a nice side effect Nx also uses `cores`